### PR TITLE
frontend dependency update

### PR DESCRIPTION
**Pull request by phoenixAja**
### Summary:
- **What:** upgrade `next` to version 12.0.2 and `webpack` to latest 5.61.0 (all `webpack` 4 versions had existing vulnerable package `set-value`), this was done to address security issues associated with both packages. Mainly I just want dependabot to feel like we hear him, and that we care about he has to say 😄 

- **Ticket:** [sc165612](https://app.shortcut.com/genepi/story/165612)
- **Env:** `https://dependency-upgrade-frontend.dev.genepi.czi.technology`

### Notes:
with this dependency upgrade to Next 12 we had to replace next-images with the next/ image component (since Next 12 supports static image imports natively and next-image is not compatible with newer version of Next understandably). This affected how we were styling png images in aspen, i had to make some tweaks to make the images match the design specs. 

Another Note: I'm going to be adding a process to update our dependencies in another PR (and i'll break up clubhouse ticket accordingly)

### Checklist:
- [x] I merged latest `trunk`
- [x] I manually verified the change
- [x] I added labels to my PR
- [ ] I tested in multiple browsers
- [ ] I added relevant unit tests
- [ ] I have notified others of changes they need to make locally (migrations, jobs, package updates, etc)

### CODE REVIEW

File: src/frontend/src/common/components/library/data_subview/components/RedirectConfirmationModal/style.ts [hunk 1]
```diff
@@ -4,16 +4,10 @@ import { P } from "src/common/styles/support/style";
...
-
-export const StyledImg = styled("img")`
-  height: 45px;
-
   ${(props) => {
     const spaces = getSpaces(props);
-
     return `
-      margin-bottom: ${spaces?.l}px;
+      margin-top: ${spaces?.l}px;
```
Comment (phoenixAja) [inline src/frontend/src/common/components/library/data_subview/components/RedirectConfirmationModal/style.ts:10]: had to add margin-top to `StyledHeader`, styling wasn't being properly applied to Next Image component and this seemed like a more straightforwards work around

File: src/frontend/src/common/components/library/data_subview/components/UsherTreeFlow/components/UsherConfirmationModal/index.tsx [hunk 2]
```diff
@@ -34,11 +34,12 @@ const UsherConfirmationModal = ({
     <div>
       <RedirectConfirmationModal
         content={content}
-        img={usherLogo as string}
+        img={usherLogo as unknown as string}
```
Comment (phoenixAja) [inline src/frontend/src/common/components/library/data_subview/components/UsherTreeFlow/components/UsherConfirmationModal/index.tsx:37]: this was to address some typehint weirdness, it didn't like converting to a string directly.

File: src/frontend/src/common/components/library/data_subview/components/RedirectConfirmationModal/index.tsx [hunk 3]
```diff
@@ -19,11 +24,18 @@ const RedirectConfirmationModal = ({
   isOpen,
   onClose,
   onConfirm,
+  logoWidth,
+  logoHeight,
```
Comment (mmmmmmaya) [inline src/frontend/src/common/components/library/data_subview/components/RedirectConfirmationModal/index.tsx]: prefer to set defaults here
```
logoHeight = 45,
```

File: src/frontend/src/common/components/library/data_subview/components/UsherTreeFlow/components/UsherConfirmationModal/index.tsx (hunk 2 above)
Comment (mmmmmmaya) [reply src/frontend/src/common/components/library/data_subview/components/UsherTreeFlow/components/UsherConfirmationModal/index.tsx:37]: This shouldn't be necessary, and the original doesn't cause type failures for me. Is there an extra tool you have in your IDE?

File: src/frontend/src/common/components/library/data_subview/components/RedirectConfirmationModal/index.tsx [hunk 4]
```diff
@@ -19,11 +24,18 @@ const RedirectConfirmationModal = ({
...
   ...props
 }: Props): JSX.Element => {
   const title = (
     <>
-      <StyledImg src={img} />
+      <Image // using next image for png files, see https://nextjs.org/docs/basic-features/image-optimization#image-sizing
+        src={img}
+        height={logoHeight || 45} // 45 px
+        width={logoWidth}
+        layout={"intrinsic"} // Scale down to fit width of container, up to image size
```
Comment (mmmmmmaya) [inline src/frontend/src/common/components/library/data_subview/components/RedirectConfirmationModal/index.tsx:37]:
```suggestion
        layout="intrinsic" // Scale down to fit width of container, up to image size
```

File: src/frontend/src/common/components/library/data_subview/components/RedirectConfirmationModal/index.tsx (hunk 3 above)
Comment (phoenixAja) [reply src/frontend/src/common/components/library/data_subview/components/RedirectConfirmationModal/index.tsx]: lol wow of course they should be there! Not really sure what i was thinking

File: src/frontend/src/common/components/library/data_subview/components/UsherTreeFlow/components/UsherConfirmationModal/index.tsx (hunk 2 above)
Comment (phoenixAja) [reply src/frontend/src/common/components/library/data_subview/components/UsherTreeFlow/components/UsherConfirmationModal/index.tsx:37]: no i got the error from running make frontend-lint and i'm pretty sure that they were affecting the build, but only after i updated the dependencies. I can try removing it to see if i'm wrong

Comment (phoenixAja) [reply src/frontend/src/common/components/library/data_subview/components/UsherTreeFlow/components/UsherConfirmationModal/index.tsx:37]: so just tested it again and it wasn't breaking the build, this is what vs code was saying, not sure if it's best to cast to unknown or keep it as casting to string directly. I also tried using `String(usherLogo)` and that fixed the issue, what do you think is the best thing to do here? 

Screen Shot 2021-11-03 at 3 56 54 PM

Comment (phoenixAja) [reply src/frontend/src/common/components/library/data_subview/components/UsherTreeFlow/components/UsherConfirmationModal/index.tsx:37]: i know String(usherLogo) is problematic if usherLogo is ever undefined

File: src/frontend/src/common/components/library/data_subview/components/RedirectConfirmationModal/index.tsx (hunk 3 above)
Comment (mmmmmmaya) [reply src/frontend/src/common/components/library/data_subview/components/RedirectConfirmationModal/index.tsx]:
```suggestion
  logoHeight = 45,
```

File: src/frontend/src/common/components/library/data_subview/components/RedirectConfirmationModal/index.tsx [hunk 5]
```diff
@@ -19,11 +24,18 @@ const RedirectConfirmationModal = ({
...
+  logoWidth,
+  logoHeight,
   ...props
 }: Props): JSX.Element => {
   const title = (
     <>
-      <StyledImg src={img} />
+      <Image // using next image for png files, see https://nextjs.org/docs/basic-features/image-optimization#image-sizing
+        src={img}
+        height={logoHeight || 45} // 45 px
```
Comment (mmmmmmaya) [inline src/frontend/src/common/components/library/data_subview/components/RedirectConfirmationModal/index.tsx]:
```suggestion
        height={logoHeight}
```

**Review by mmmmmmaya — APPROVED** [review]

**Pull request merged**